src/js/ui/shared/form.ts

```diff
@@ -13,6 +14,21 @@ function renderOption(label: string, value: string, storedValue: string, labelAr
   }, i18n(label, labelArg, labelArg2));
 }
 
+
+function renderOptionGroup(label:string, value:string | SelectOptionGroup, storedValue:string, labelArg:string, labelArg2:string): Mithril.Children {
+  if (typeof value === 'string') {
```

Comment:
Looks like typeGuard can be implemented with a function or simply allowing typescript to infer types from the `typeof` if statements. I removed the manual casting and the latter approach seems to work great. Good to know.